src/main/kotlin/com/sourcegraph/cody/chat/ui/MessagePart.kt

```diff
@@ -59,9 +61,21 @@ class CodeEditorPart(
     }
   }
 
-  private fun updateText(project: Project, text: String) {
-    this.text = text
-    WriteCommandAction.runWriteCommandAction(
-        project, Computable { editor.document.replaceText(text, System.currentTimeMillis()) })
+  private fun updateText(project: Project, newText: String) {
+    val newIndex = newText.length - 1
+    if (newIndex >= lastIndex) {
+      val size = newIndex - lastIndex
```

Comment:
I'm not sure if that is correct.
I think - but I'm not 100% sure - I have seen LLM responses when it was retracting few last tokens and replacing them with something else. So to be really sure we should also compare content, not just the length.

Reply:
Updated. Please take a look.